types/algoliasearch/index.d.ts

```diff
@@ -1337,81 +1339,88 @@ declare namespace algoliasearch {
     /**
      * Specify the list of approximation that should be considered as an exact match in the ranking formula
      * default: ['ignorePlurals', 'singleWordSynonym']
      * 'ignorePlurals': alternative words added by the ignorePlurals feature
      * 'singleWordSynonym': single-word synonym (For example "NY" = "NYC")
      * 'multiWordsSynonym': multiple-words synonym
-     * https://github.com/algolia/algoliasearch-client-js#alternativesasexact
+     * https://www.algolia.com/doc/api-reference/api-parameters/alternativesAsExact/
      */
-    alternativesAsExact?: any;
+    alternativesAsExact?: string[];
     /**
      * If set to 1, enables the distinct feature, disabled by default, if the attributeForDistinct index setting is set.
-     * https://github.com/algolia/algoliasearch-client-js#distinct
+     * https://www.algolia.com/doc/api-reference/api-parameters/distinct/
      */
-    distinct?: any;
+    distinct?: number|boolean;
     /**
      * If set to true, the result hits will contain ranking information in the _rankingInfo attribute.
      * default: false
-     * https://github.com/algolia/algoliasearch-client-js#getrankinginfo
+     * https://www.algolia.com/doc/api-reference/api-parameters/getRankingInfo/
      */
     getRankingInfo?: boolean;
     /**
+     * @deprecated Use `numericAttributesForFiltering` instead
      * All numerical attributes are automatically indexed as numerical filters
      * default: ''
-     * https://github.com/algolia/algoliasearch-client-js#numericattributestoindex
+     * https://www.algolia.com/doc/api-reference/api-parameters/numericAttributesForFiltering/
      */
     numericAttributesToIndex?: string[];
+    /**
+     * All numerical attributes are automatically indexed as numerical filters
+     * default: ''
+     * https://www.algolia.com/doc/api-reference/api-parameters/numericAttributesForFiltering/
+     */
+    numericAttributesForFiltering?: string[];
     /**
      * @deprecated please use filters instead
      * A string that contains the comma separated list of numeric filters you want to apply.
-     * https://github.com/algolia/algoliasearch-client-js#numericfilters-deprecated
+     * https://www.algolia.com/doc/api-reference/api-parameters/numericFilters/
      */
     numericFilters?: string[];
     /**
-     * @deprecated
      * Filter the query by a set of tags.
-     * https://github.com/algolia/algoliasearch-client-js#tagfilters-deprecated
+     * Default: []
+     * https://www.algolia.com/doc/api-reference/api-parameters/tagFilters/
      */
-    tagFilters?: string;
+    tagFilters?: string[];
     /**
-     * @deprecated
      * Filter the query by a set of facets.
-     * https://github.com/algolia/algoliasearch-client-js#facetfilters-deprecated
+     * Default: []
+     * https://www.algolia.com/doc/api-reference/api-parameters/facetFilters/
      */
-    facetFilters?: string | string[]
+    facetFilters?: string[]|string[][];
```

Comment:
technically also accepts string form, but that should be done in `filters` instead

Reply:
It's described in the REST API docs, but since it's not documented anywhere else, I think your choice here is the correct one